MathLib/LinAlg/Eigen/EigenLinearSolver.cpp

```diff
@@ -0,0 +1,145 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "EigenLinearSolver.h"
+
+#include <logog/include/logog.hpp>
+#include <unsupported/Eigen/IterativeSolvers>
+
+#include "EigenVector.h"
+#include "EigenMatrix.h"
+#include "EigenTools.h"
+
+namespace MathLib
+{
+
+using boost::property_tree::ptree;
+
+namespace details
+{
+
+template <class T_SOLVER>
+class EigenDirectSolver : public EigenLinearSolver::IEigenSolver
+{
+public:
+    EigenDirectSolver(EigenMatrix::RawMatrixType &A) : _A(A)
+    {
+        // fill A and b;
+        // Compute the ordering permutation vector from the structural pattern of A
+        _solver.analyzePattern(A);
+        // Compute the numerical factorization
+        _solver.factorize(A);
+        if(_solver.info()!=Eigen::Success) {
+            INFO("\t failed");
+            return;
+        }
+    }
+
+    virtual ~EigenDirectSolver() {}
+
+    void solve(EigenVector::RawVectorType &b, EigenVector::RawVectorType &x, EigenOption &) override
+    {
+        //Use the factors to solve the linear system
+        INFO("-> solve");
+        x = _solver.solve(b);
+    }
+
+private:
+    T_SOLVER _solver;
```

Comment:
Is EigenIterativeSolver owning the _solver or just using it?
Precaution to avoid copies of _solver.